src/controllers/Interactions.ts

```diff
@@ -0,0 +1,81 @@
+import { Interaction, TextChannel } from "discord.js";
+import { DateTime } from "luxon";
+import LeaderboardRepository from "../repositories/LeaderboardRepository";
+import QueueRepository from "../repositories/QueueRepository";
+import { BallChaser } from "../types/common";
+import MessageBuilder from "../utils/MessageBuilder";
+
+export async function getCurrentQueue(queueChannel: TextChannel): Promise<void> {
+  
+  const ballchasers = await QueueRepository.getAllBallChasersInQueue();
+
+  await queueChannel.send({ 
+    components: [MessageBuilder.queueButtons],
+    embeds: [MessageBuilder.queueMessage(ballchasers)]
+  });
+}
+
+export async function onInteraction(buttonInteraction: Interaction, queueChannel: TextChannel): Promise<void> {
+
+  if (!buttonInteraction.isButton()) return;
+  await buttonInteraction.deferReply();
+
+  switch (buttonInteraction.customId) {
+    case "joinQueue": {
+
+      const queueMember = await QueueRepository.getBallChaserInQueue(buttonInteraction.user.toString());
```

Comment:
Instead of `buttonInteraction.user.toString()` can we use `buttonInteraction.user.id`, please? Doing the `toString` includes the `<@>` around the actual ID which makes storing it in the DB weird.

Reply:
Resolved.